### PR TITLE
Magento 2.3.4 backward incompatible change

### PR DESCRIPTION
**Pull request by jfrontain**
This pull request (PR)  adds information about a backward incompatibility in Magento 2.3.4 to the backward incompatible topic and adds links from the release notes to the bic topic. 

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.3/release-notes/backward-incompatible-changes/index.html

https://devdocs.magento.com/guides/v2.3/release-notes/release-notes-2-3-4-commerce.html

https://devdocs.magento.com/guides/v2.3/release-notes/release-notes-2-3-4-open-source.html

### CODE REVIEW

File: src/guides/v2.3/release-notes/backward-incompatible-changes/index.md [hunk 1]
```diff
@@ -53,6 +53,43 @@ The following example shows the diff in `Magento\Catalog\Model\ResourceModel\Pro
...
+</config>
+```
+
+Deployments that use custom customer sections with this type of action invalidation may face an issue when private content isn’t entered correctly. As a result, problems may occur in the storefront checkout and cart workflows.
+
+### Workaround
+
+For all actions where it is required to invalidate custom customer sections, use either `*` as the section name or use empty actions and ensure that they will not be overridden by any other rules.
+
+For all actions where it is required to invalidate custom customer sections, use either `*` as the section name or use empty actions and ensure that they will not be overridden by any other rules.
```
Comment (keharper) [inline src/guides/v2.3/release-notes/backward-incompatible-changes/index.md]: Delete this duplicate line.

File: src/guides/v2.3/release-notes/backward-incompatible-changes/index.md [hunk 2]
```diff
@@ -53,6 +53,43 @@ The following example shows the diff in `Magento\Catalog\Model\ResourceModel\Pro
...
+```xml
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Customer:etc/sections.xsd">
+    <action name="customer/account/editPost">
+        <section name="*"/>
+    </action>
+</config>
+```
+
+See [Private content](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/cache/page-caching/private-content.html) for information on how to invalidate private content.
```
Comment (keharper) [inline src/guides/v2.3/release-notes/backward-incompatible-changes/index.md]:
```suggestion
See [Private content]({{page.baseurl}}/extension-dev-guide/cache/page-caching/private-content.html) for information on how to invalidate private content.
```

File: src/guides/v2.3/release-notes/backward-incompatible-changes/index.md [hunk 3]
```diff
@@ -53,6 +53,43 @@ The following example shows the diff in `Magento\Catalog\Model\ResourceModel\Pro
         );
 ...
 ```
+## Customer data sections invalidation
+
+Magento 2.3.4 introduced a change in the customer data sections invalidation logic.
```
Comment (keharper) [inline src/guides/v2.3/release-notes/backward-incompatible-changes/index.md]:
```suggestion
Magento 2.3.4 introduced a change in the invalidation logic of customer data sections.
```

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 4]
```diff
@@ -2160,6 +2160,10 @@ We have fixed hundreds of issues in the Magento 2.3.4 core code.
 
 *  The WYSIWYG editor now saves quotation marks correctly. Previously, quotation marks were converted to `&quot;`.
 
+## Known issue
+
+This release introduces an enhancement to the  customer data sections invalidation mechanism logic that is not compatible with how pre-2.3.4 deployments.  You can no longer invalidate custom customer sections in etc/frontend/sections.xml by declaring an action node without specifying any related sections. See [Magento 2.3 backward incompatible changes](https://devdocs.magento.com/guides/v2.3/release-notes/backward-incompatible-changes/index.html) for a discussion of this change.
```
Comment (keharper) [inline src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]:
```suggestion
This release introduces an enhancement to the invalidation logic for customer data sections that is not compatible with pre-2.3.4 deployments.  You can no longer invalidate custom customer sections in `etc/frontend/sections.xml` files by declaring an action node without specifying any related sections. See [Magento 2.3 backward incompatible changes]({{page.baseurl}}/release-notes/backward-incompatible-changes/index.html) for a discussion of this change.
```

File: src/guides/v2.3/release-notes/release-notes-2-3-4-open-source.md [hunk 5]
```diff
@@ -1818,6 +1818,10 @@ We have fixed hundreds of issues in the Magento 2.3.4 core code.
 
 *  The WYSIWYG editor now saves quotation marks correctly. Previously, quotation marks were converted to `&quot;`.
 
+## Known issue
+
+This release introduces an enhancement to the  customer data sections invalidation mechanism logic that is not compatible with how pre-2.3.4 deployments.  You can no longer invalidate custom customer sections in etc/frontend/sections.xml by declaring an action node without specifying any related sections. See [Magento 2.3 backward incompatible changes](https://devdocs.magento.com/guides/v2.3/release-notes/backward-incompatible-changes/index.html) for a discussion of this change.
```
Comment (keharper) [inline src/guides/v2.3/release-notes/release-notes-2-3-4-open-source.md]:
```suggestion
This release introduces an enhancement to the invalidation logic for customer data sections that is not compatible with pre-2.3.4 deployments.  You can no longer invalidate custom customer sections in `etc/frontend/sections.xml` files by declaring an action node without specifying any related sections. See [Magento 2.3 backward incompatible changes]({{page.baseurl}}/release-notes/backward-incompatible-changes/index.html) for a discussion of this change.
```

File: src/guides/v2.3/release-notes/backward-incompatible-changes/index.md (hunk 1 above)
Comment (jfrontain) [reply src/guides/v2.3/release-notes/backward-incompatible-changes/index.md]: done

File: src/guides/v2.3/release-notes/backward-incompatible-changes/index.md (hunk 2 above)
Comment (jfrontain) [reply src/guides/v2.3/release-notes/backward-incompatible-changes/index.md]: changed in source file

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md (hunk 4 above)
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: changed in source

File: src/guides/v2.3/release-notes/release-notes-2-3-4-open-source.md (hunk 5 above)
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-open-source.md]: changed in source

File: src/guides/v2.3/release-notes/backward-incompatible-changes/index.md (hunk 3 above)
Comment (jfrontain) [reply src/guides/v2.3/release-notes/backward-incompatible-changes/index.md]: changed in source

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 6]
```diff
@@ -2160,6 +2160,10 @@ We have fixed hundreds of issues in the Magento 2.3.4 core code.
 
 *  The WYSIWYG editor now saves quotation marks correctly. Previously, quotation marks were converted to `&quot;`.
 
+## Known issue
+
+This release introduces an enhancement to the invalidation logic for customer data sections that is not compatible with how pre-2.3.4 deployments. You can no longer invalidate custom customer sections in `etc/frontend/sections.xml` files by declaring an action node without specifying any related sections. See [Magento 2.3 backward incompatible changes]({{page.baseurl}}/release-notes/backward-incompatible-changes/index.html) for a discussion of this change.
```
Comment (keharper) [inline src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]:
```suggestion
This release introduces an enhancement to the invalidation logic for customer data sections that is not compatible with pre-2.3.4 deployments. You can no longer invalidate custom customer sections in `etc/frontend/sections.xml` files by declaring an action node without specifying any related sections. See [Magento 2.3 backward incompatible changes]({{page.baseurl}}/release-notes/backward-incompatible-changes/index.html) for a discussion of this change.
```

File: src/guides/v2.3/release-notes/release-notes-2-3-4-open-source.md [hunk 7]
```diff
@@ -1818,6 +1818,10 @@ We have fixed hundreds of issues in the Magento 2.3.4 core code.
 
 *  The WYSIWYG editor now saves quotation marks correctly. Previously, quotation marks were converted to `&quot;`.
 
+## Known issue
+
+This release introduces an enhancement to the invalidation logic for customer data sections that is not compatible with how pre-2.3.4 deployments.  You can no longer invalidate custom customer sections in `etc/frontend/sections.xml` files by declaring an action node without specifying any related sections. See [Magento 2.3 backward incompatible changes]({{page.baseurl}}/release-notes/backward-incompatible-changes/index.html) for a discussion of this change.
```
Comment (keharper) [inline src/guides/v2.3/release-notes/release-notes-2-3-4-open-source.md]:
```suggestion
This release introduces an enhancement to the invalidation logic for customer data sections that is not compatible with pre-2.3.4 deployments.  You can no longer invalidate custom customer sections in `etc/frontend/sections.xml` files by declaring an action node without specifying any related sections. See [Magento 2.3 backward incompatible changes]({{page.baseurl}}/release-notes/backward-incompatible-changes/index.html) for a discussion of this change.
```

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md (hunk 6 above)
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: entered in source

**Comment by jfrontain** [comment]
running tests

**Pull request merged**